frontend/src/lib/components/CompareFilter/CompareFilter.tsx

```diff
@@ -1,30 +1,117 @@
-import { LemonCheckbox } from '@posthog/lemon-ui'
+import { LemonSelect } from '@posthog/lemon-ui'
 import { useActions, useValues } from 'kea'
+import { RollingDateRangeFilter } from 'lib/components/DateFilter/RollingDateRangeFilter'
+import { dateFromToText } from 'lib/utils'
+import { useState } from 'react'
 import { insightLogic } from 'scenes/insights/insightLogic'
 import { insightVizDataLogic } from 'scenes/insights/insightVizDataLogic'
 
 export function CompareFilter(): JSX.Element | null {
     const { insightProps, canEditInsight } = useValues(insightLogic)
 
-    const { compare, supportsCompare } = useValues(insightVizDataLogic(insightProps))
+    const { compare, supportsCompare, compareTo } = useValues(insightVizDataLogic(insightProps))
     const { updateInsightFilter } = useActions(insightVizDataLogic(insightProps))
 
+    // This keeps the state of the rolling date range filter, even when different drop down options are selected
+    // The default value for this is one month
+    const [compareToUi, setCompareToUi] = useState<string>(compareTo || '-1m')
+
     const disabled: boolean = !canEditInsight || !supportsCompare
 
     // Hide compare filter control when disabled to avoid states where control is "disabled but checked"
     if (disabled) {
         return null
     }
 
+    if (!!compareTo && compareToUi != compareTo) {
+        setCompareToUi(compareTo)
+    }
+
+    const options = [
+        {
+            value: 'none',
+            label: 'No comparison between periods',
+        },
+        {
+            value: 'previous',
+            label: 'Compare to previous period',
+        },
+        {
+            value: 'compareTo',
+            label: (
+                <div onClick={() => updateInsightFilter({ compare: true, compareTo: compareToUi })}>
```

Comment:
Seems non-typical for an `onClick` to be on a div, maybe `LemonSelect`'s `onSelect` prop could do this job here?